### ISSUE TITLE
Implement triggers

### ISSUE THREAD

**Issue by skobyda** [skobyda/IDS#2]
Trigger sa využíva v prípade, keď pri vykonaní jednej akcie chceme vykonať inú akciu.
Pravidlo pri tvorbe triggerov je, že sa využívajú len v prípade, keď ich treba využíť.
Používa sa ale v prípadoch, kedy treba overiť niečo počas behu? Napríklad prípad, ktorý naošetruje náš SQL skript:
Pri vytvorení novej rezervácie zabezpečiť, či už nebola preročená kapacita sály.

**Comment by MichalZelenak** [comment]
no ale tie rezervácie sa predsa vytvárajú na sedadlo... ak už nie je sedadlo bez rezervácie tak by nemalo ísť ani vytvoriť rezerváciu dalšiu...možno by bolo dobré k sedadlu pridať atribút obsadenosti

**Comment by MichalZelenak** [comment]
zároveň v zadaní toho multikina je napísané že kapacitu sály nemáme uvažovať... no či to platí aj pre ids to už neviem

**Comment by skobyda** [comment]
Dobre, ale narozdiel od tohto príkladu, chcem používať triggery hlavne na validáciu podobných prípadov. Napadá ťa ešte niečo, k čomu by sa mohli triggeri použiť?

**Comment by MichalZelenak** [comment]
číslo účtu by sa malo dať nejako validovať asi... keďže tam sú dve písmená prvé a potom čísla....ak teda uvažujeme iban...a pri rezevácií online overiť vek klienta voči vekovému obmedzeniu

**Comment by MichalZelenak** [comment]
nie som si istý... ale dnes ma napadlo či by sa nemala tým triggerom riešiť aj obsadenosť sedadla k danej projekcii pri zakúpení vstupenky

**Comment by skobyda** [comment]
Ano, takéto veci som myslel. Celkovo tým by som robil runtime validáciu. Teda veci, ktoré sa dajú overiť až za behu, a nie pri tvorbe/návrhu databázy, pretože samozrejme pri tvorbe/návrhu databázy obsadenosť ešte nemá žiadnu hodnotu, tak vtedy to nemáme ako validovať.

Takže dobrá pripomienka, napadajú ťa ďaľšie veci ktoré sa dajú validovať až za runtimu?
